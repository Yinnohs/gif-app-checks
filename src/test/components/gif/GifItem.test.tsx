import { describe, it, expect } from "vitest";
import { GifItem } from "../../../components";
import { render, screen } from "@testing-library/react";

const testingGif: GiftItem = {
  id: "22",
  url: "https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fscholartree.ca%2Fblog%2Fwp-content%2Fuploads%2F2018%2F09%2Fhomer_simpson_rat_testing.gif&f=1&nofb=1&ipt=39d79f305b2235a683a8c5d7a8c32d88e5c2529e12f028d2352dcfb70011038b&ipo=images",
  title: "testing gif",
};

describe("Testing the GifItem component", () => {
  it("Should be the same as the snapshot", () => {
    const { container } = render(<GifItem gif={testingGif} />);
    expect(container).toMatchSnapshot();
  });

  it("Should shown the correct image alt and url indicated", () => {
    const { url, title } = testingGif;
    render(<GifItem gif={testingGif} />);
    const img = screen.getByRole("img");
    expect(img.getAttribute("src")).toEqual(url);
    expect(img.getAttribute("alt")).toEqual(title);
  });
});
