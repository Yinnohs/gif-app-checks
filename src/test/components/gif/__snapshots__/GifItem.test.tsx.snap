// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the GifItem component > Should be the same as the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="testing gif"
      src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fscholartree.ca%2Fblog%2Fwp-content%2Fuploads%2F2018%2F09%2Fhomer_simpson_rat_testing.gif&f=1&nofb=1&ipt=39d79f305b2235a683a8c5d7a8c32d88e5c2529e12f028d2352dcfb70011038b&ipo=images"
    />
    ;
    <p>
      testing gif
    </p>
  </div>
</div>
`;
